### PR TITLE
Feature/save user agreement

### PR DESCRIPTION
**Pull request by jschaul**
This pull request implements:
* saving of the user agreement (Accepting terms and conditions) to LDAP (a boolean & a date). 

Also, the in-memory implementation and the LDAP implementation for the data backend have been streamlined a little, and there is now the functionality of running integration tests (via `it:test`) separately, but one can share source files and test code (see the build.sbt). 

See https://github.com/yetu/oauth2-provider/blob/693d8f9c88cc033c277963824d0907438ed1d6d7/test/com/yetu/oauth2provider/services/data/UserServiceSpec.scala for an example. All tests whose file names end in `...ITSpec` are executed upon `it:test`, and all others execute upon `test`. 

@ElliDy @dev-tim @mr-mig

### CODE REVIEW

**Comment by ElliDy** [comment]
Joé wrote me an email, what I can still do to complete this pull request:
1) no global vars,
2) delete frontend validation code
3) replace Assets.at(...) trough Assets.versioned(...) to solve cache problems

I would propose to make the last two points in the setup-flow branch, because the code for the pages changed there and so I found this a better place for it. I will start with this no.
For the first one I´m not sure, what it concretely means. So I will wait for Joé to explain it ;)

I will start as well reviewing now and give feedback. 

The branch setup-flow has to be rebased after this one is merged.

File: app/com/yetu/oauth2provider/oauth2/models/YetuUser.scala [hunk 1]
```diff
@@ -23,15 +24,16 @@ case class YetuUser(identityId: IdentityId, uid: String, firstName: String, last
     registrationDate: Option[Date] = None,
     contactInfo: Option[ContactInfo] = None,
     imageUrl: Option[String] = None,
-    publicKey: Option[YetuPublicKey] = None) {
+    publicKey: Option[YetuPublicKey] = None,
+    userAgreement: Option[UserAgreement] = None) {
 
   //TODO: clean this up.
```
Comment (ElliDy) [inline app/com/yetu/oauth2provider/oauth2/models/YetuUser.scala]: What do we need to clean up here exactely?

Comment (jschaul) [inline app/com/yetu/oauth2provider/oauth2/models/YetuUser.scala]: At the moment, securesocial expects a "BasicProfile" when doing login/registration. But there is also a YetuUser, as we wish to store a few extra fields, like e.g. the public SSH key. So then there are conversions between YetuUser and BasicProfile back and forth, and with this commit we also changed the BasicProfile itself. So if we do that anyway, we do not need to maintain two different kinds of "User" Types that are mostly the same but slightly different for historic users. That should be cleaned up, either by removing one case class, or by extracting additional fields into a separate class.

Comment (ElliDy) [inline app/com/yetu/oauth2provider/oauth2/models/YetuUser.scala]: Should we do it now or later?

**Comment by jschaul** [comment]
Comments still welcome.

**Pull request merged**